src/Transport/Sending/DefaultBatchedOperationsToBrokeredMessagesConverter.cs

```diff
@@ -81,8 +80,16 @@ void SetReplyToAddress(OutgoingMessage outgoingMessage, BrokeredMessage brokered
 
                     if (selected != null)
                     {
-                        replyTo = mapper.Map(new EntityAddress(replyTo.Name, selected.Name));
+                        if (mapper is DefaultNamespaceNameToConnectionStringMapper)
```

Comment:
I didn't like it either, I wanted to check on the boolean that indicated whether UseNamespaceNames is on or not, but that config method registers a type (so I checked on that type instead)

The logic:
if 'UseName' is on -> always append a name
if 'UseName' is off -> always append a connectionstring